pkg/cmd/repo/rename/rename.go

```diff
@@ -0,0 +1,165 @@
+package rename
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/context"
+	"github.com/cli/cli/v2/git"
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/pkg/prompt"
+	"github.com/spf13/cobra"
+)
+
+type RenameOptions struct {
+	HttpClient      func() (*http.Client, error)
+	IO              *iostreams.IOStreams
+	Config          func() (config.Config, error)
+	BaseRepo        func() (ghrepo.Interface, error)
+	Remotes         func() (context.Remotes, error)
+	DoConfirm       bool
+	HasRepoOverride bool
+	newRepoSelector string
+}
+
+func NewCmdRename(f *cmdutil.Factory, runf func(*RenameOptions) error) *cobra.Command {
+	opts := &RenameOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Remotes:    f.Remotes,
+		Config:     f.Config,
+	}
+
+	var confirm bool
+
+	cmd := &cobra.Command{
+		Use:   "rename [<new-name>]",
+		Short: "Rename a repository",
+		Long: heredoc.Doc(`Rename a GitHub repository
+
+		By default, renames the current repository otherwise renames the specified repository.`),
```

Comment:
Nit: the whole reason we're using heredoc-like strings here is so we can indent them in code. We should take advantage of that and format this in a more readable way:
```suggestion
		Long: heredoc.Doc(`
			Rename a GitHub repository

			By default, this renames the current repository; otherwise renames the specified repository.
		`),
```
Also note the fixed punctuation in the code suggestion.